src/python/doc/wasserstein_distance_user.rst

```diff
@@ -84,3 +90,80 @@ The output is:
     point 1 in dgm1 is matched to point 2 in dgm2
     point 2 in dgm1 is matched to the diagonal
     point 1 in dgm2 is matched to the diagonal
+
+
+Barycenters
+-----------
+
+A Frechet mean (or barycenter) is a generalization of the arithmetic 
+mean in a non linear space such as the one of persistence diagrams.  
+Given a set of persistence diagrams :math:`\mu_1 \dots \mu_n`, it is 
+defined as a minimizer of the variance functional, that is of        
+:math:`\mu \mapsto \sum_{i=1}^n d_2(\mu,\mu_i)^2`.                   
+where :math:`d_2` denotes the Wasserstein-2 distance between         
+persistence diagrams.                                                
+It is known to exist and is generically unique. However, an exact    
+computation is in general untractable. Current implementation        
+available is based on (Turner et al., 2014), 
+:cite:`turner2014frechet`
+and uses an EM-scheme to 
+provide a local minimum of the variance functional (somewhat similar 
+to the Lloyd algorithm to estimate a solution to the k-means         
+problem). The local minimum returned depends on the initialization of
+the barycenter.                                                      
+The combinatorial structure of the algorithm limits its              
+performances on large scale problems (thousands of diagrams and of points 
+per diagram).                                                        
+
+.. figure::                                                    
+     ./img/barycenter.png                                      
+     :figclass: align-center                                   
+                                                               
+     Illustration of Frechet mean between persistence          
+     diagrams.                                                 
+
+
+.. autofunction:: gudhi.barycenter.lagrangian_barycenter
```

Comment:
Missing `wasserstein`, it doesn't show up in the doc.

Reply:
Done